photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,446 @@
+        psf : `astropy.modeling.Fittable2DModel` instance
+            PSF or PRF model to fit the data. Could be one of the models in
+            this package like `~photutils.psf.sandbox.DiscretePRF`,
+            `~photutils.psf.IntegratedGaussianPRF`, or any other suitable
+            2D model.
+            This object needs to identify three parameters (position of
+            center in x and y coordinates and the flux) in order to set them
+            to suitable starting values for each fit. The names of these
+            parameters can be given as follows:
+
+            - Set ``psf.psf_xname``, ``psf.psf_yname`` and
+              ``psf.psf_fluxname`` to strings with the names of the respective
+              psf model parameter.
+            - If those attributes are not found, the names ``x_0``, ``y_0``
+              and ``flux`` are assumed.
+
+            `~photutils.psf.prepare_psf_model` can be used to prepare any 2D
+            model to match these assumptions.
+        fitshape : array-like
+            Rectangular shape around the center of a star which will be used
+            to collect the data to do the fitting, e.g. (5, 5) means to take
+            the following relative pixel positions: [-2, -1, 0, 1, 2].
+            Also, each element of ``fitshape`` must be an odd number.
+        fitter : Fitter instance (default=LevMarLSQFitter())
+            Fitter object used to compute the optimized centroid positions
+            and/or flux of the identified sources. See
+            `~astropy.modeling.fitting` for more details on fitters.
+        niters : int (default=3)
+            Number of iterations to perform the loop FIND, GROUP, SUBTRACT,
+            NSTAR.
+        aperture_radius : float (default=None)
+            The radius used to compute initial estimates for the fluxes of
+            sources. If ``None``, one fwhm will be used. 
+
+        Notes
+        -----
+        If there are problems with fitting large groups, change the parameters
+        of the grouping algorithm to reduce the number of sources in each
+        group or input a ``star_groups`` table that only includes the groups
+        that are relevant (e.g. manually remove all entries that coincide with
+        artifacts).
+
+        References
+        ----------
+        [1] Stetson, Astronomical Society of the Pacific, Publications,
+            (ISSN 0004-6280), vol. 99, March 1987, p. 191-222.
+            Available at: http://adsabs.harvard.edu/abs/1987PASP...99..191S
+        """
+
+        self.find = find
+        self.group = group
+        self.bkg = bkg
+        self.psf = psf
+        self.fitter = fitter
+        self.niters = niters
+        self.fitshape = fitshape
+        self.aperture_radius = aperture_radius
+
+        @property
+        def niters(self):
+            return self._niters
+
+        @niters.setter
+        def niters(self, niters):
+            if isinstance(niters, int) and niters > 0:
```

Comment:
I don't think you want to use `isinstance` here.  `isinstance` on numeric types tends to have unintended side-effects.  For example, in Python 2 "long integers" are a different type from "regular" integers.  So ``isinstance(2L, int)`` evaluates to False, but it should be reasonable to use as `niters`.  Similarly, if I accidentally pass in a float that has an integer value (e.g. 4.0) it should be acceptable. So instead I'd suggest something like this:

```
def niters(self, niters):
    try:
        niters = int(niters)
    except:
        raise ValueError('niters must be an integer or convertable into an integer')
    if niters<=0:
        raise ValueError('niters must be positive')
    self._niters = niters
```
That way *anything* that can be successfully cast into an int will work.